test/NuGet.Core.FuncTests/Dotnet.Integration.Test/ProjectFileBuilder.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Dotnet.Integration.Test;
+
+namespace NuGet.Test.Utility
+{
+    /// <summary>
+    /// Represents a builder for dotnet core projects for integration tests.
+    /// Relies on <c>MsbuildIntegrationTestFixture</c> to programmatically create the project
+    /// </summary>
+    /// <seealso cref="MsbuildIntegrationTestFixture"/>
+    public class ProjectFileBuilder
+    {
+        public string PackageIcon { get; private set; }
+        public string PackageIconUrl { get; private set; }
+        public string ProjectName { get; private set; }
+        public List<Tuple<string, string, string>> ItemGroupEntries { get; private set; }
+        public string BaseDir { get; private set; }
+        public string ProjectFilepath => Path.Combine(BaseDir, ProjectName, $"{ProjectName}.csproj");
+        public string ProjectFolder => Path.Combine(BaseDir, ProjectName);
+        public Dictionary<string, string> Properties { get; private set; }
+
+        private ProjectFileBuilder()
+        {
+            ItemGroupEntries = new List<Tuple<string, string, string>>();
+            Properties = new Dictionary<string, string>();
+        }
+
+        /// <summary>
+        /// Factory method
+        /// </summary>
+        /// <returns>An instance of <c>ProjectFileBuilder</c></returns>
+        public static ProjectFileBuilder Create()
+        {
+            return new ProjectFileBuilder();
+        }
+
+        public void Build(MsbuildIntegrationTestFixture fixture, string path)
+        {
+            BaseDir = path;
+
+            fixture.CreateDotnetNewProject(path, ProjectName, " classlib");
+
+            ModifyProjectFile();
+        }
+
+        public ProjectFileBuilder WithItem(string type, string path, string packagePath)
```

Comment:
the second parameter, `path`, needs a better name to understand how it's different to `packagePath`

Reply:
Renamed method signature to:
`public ProjectFileBuilder WithItem(string itemType, string itemPath, string packagePath)`